stored_responses/stored_responses.go

```diff
@@ -147,11 +147,46 @@ func ProcessStoredResponses(ctx context.Context, requestJson []byte, storedRespF
 		bidderImpIdReplaceImp := flipMap(impBidderReplaceImp)
 
 		impIdToStoredResp, impBidderToStoredBidResponse := buildStoredResponsesMaps(storedResponses, impBidderToStoredBidResponseId, impIdToRespId)
+
+		errs = validateStoredAuctionResponses(impIdToStoredResp, impIdToRespId)
+		if len(errs) > 0 {
+			return nil, nil, nil, errs
+		}
+
+		errs = validateStoredBidResponses(impBidderToStoredBidResponse, impBidderToStoredBidResponseId)
+		if len(errs) > 0 {
+			return nil, nil, nil, errs
+		}
+
 		return impIdToStoredResp, impBidderToStoredBidResponse, bidderImpIdReplaceImp, nil
 	}
 	return nil, nil, nil, nil
 }
 
+func validateStoredAuctionResponses(impIdToStoredResp ImpsWithBidResponses, impIdToRespId ImpsWithAuctionResponseIDs) []error {
+	var errs []error
+	for impId, data := range impIdToStoredResp {
+		if len(data) == 0 {
+			respId := impIdToRespId[impId]
+			errs = append(errs, fmt.Errorf("failed to fetch stored auction response for impId = %s and storedAuctionResponse id = %s", impId, respId))
+		}
+	}
+	return errs
+}
+
+func validateStoredBidResponses(impBidderToStoredBidResponse ImpBidderStoredResp, impBidderToStoredBidResponseId ImpBiddersWithBidResponseIDs) []error {
```

Comment:
I'm trying to think of a way to shorten the variable names. I'm thinking you could remove the word bid from both of these variables and shorten the word response to be resp, so they'd change to `impBidderToStoredResp` and `impBidderToStoredRespId`. And we can leverage the name of the function to know that the variables passed in are stored bid responses. What do you think?

Reply:
Agree, modified parameters names. About functions names those are two different: `validateStoredBidResponses` and `validateStoredAuctionResponses`. I'm open to rename it something that looks better.